celery/backends/database/models.py

```diff
@@ -30,18 +30,34 @@ class Task(ResultModelBase):
     date_done = sa.Column(sa.DateTime, default=datetime.utcnow,
                           onupdate=datetime.utcnow, nullable=True)
     traceback = sa.Column(sa.Text, nullable=True)
+    name = sa.Column(sa.String(155))
+    args = sa.Column(sa.String(155))
+    kwargs = sa.Column(sa.String(155))
+    worker = sa.Column(sa.String(155))
+    retries = sa.Column(sa.Integer)
+    queue = sa.Column(sa.String(155))
 
     def __init__(self, task_id):
         self.task_id = task_id
 
     def to_dict(self):
-        return {
+        results = {
             'task_id': self.task_id,
             'status': self.status,
             'result': self.result,
             'traceback': self.traceback,
             'date_done': self.date_done,
         }
+        if self.app.conf.find_value_for_key('extended', 'result'):
+            results.update(
+                {'name': self.task},
```

Comment:
Why are you creating a dictionary for each key?

Reply:
@thedrow I didn't mean to lol. I just meant to add them to the existing dict. I will fix that.

Reply:
Would this work better?
```
        if self.app.conf.find_value_for_key('extended', 'result'):
            results.update(
                {
                    'name': self.task,
                    'args': self.args,
                    'kwargs': self.kwargs,
                    'worker': self.worker,
                    'retries': self.retries,
                    'queue': self.queue,
                }
            )
```

Reply:
Yes that should work better.